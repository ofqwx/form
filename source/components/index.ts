export { FormContext } from './Form/FormProvider';
export { default as FormProvider } from './Form/FormProvider';
export { default as Field } from './Field/Field';
export { default as Form } from './Form/Form';
