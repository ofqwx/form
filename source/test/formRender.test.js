/**
 * @jest-environment jsdom
 */

import React from 'react';
import { Form, Input } from '../components';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

test('Render form with initial value', () => {
  const { container } = render(
    <Form initialValues={{ firstName: 'Antonio' }} onSubmit={() => undefined}>
      <Input label="First name" name="firstName" />

      <button type="submit">Submit</button>
    </Form>
  );

  expect(container).toMatchSnapshot();
});
