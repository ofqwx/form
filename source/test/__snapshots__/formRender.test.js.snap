// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render form with initial value 1`] = `
<div>
  <form>
    <label>
      First name
    </label>
    <br />
    <input
      name="firstName"
      type="text"
      value="Antonio"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
